### ISSUE TITLE
Error while search: OperationCanceled in FindReplace.FindAll

### ISSUE THREAD

**Issue by KirillOsenkov** [mono/monodevelop#5997]
Probably need to catch OperationCanceled in FindReplace.FindAll:

```
ERROR [2018-09-16 20:37:49Z]: Error while search
System.OperationCanceledException: The operation was canceled.
  at System.Threading.Tasks.Parallel.ForEachWorker[TSource,TLocal] (System.Collections.Generic.IEnumerable`1[T] source, System.Threading.Tasks.ParallelOptions parallelOptions, System.Action`1[T] body, System.Action`2[T1,T2] bodyWithState, System.Action`3[T1,T2,T3] bodyWithStateAndIndex, System.Func`4[T1,T2,T3,TResult] bodyWithStateAndLocal, System.Func`5[T1,T2,T3,T4,TResult] bodyWithEverything, System.Func`1[TResult] localInit, System.Action`1[T] localFinally) [0x00058] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/Tasks/Parallel.cs:2283 
  at System.Threading.Tasks.Parallel.ForEach[TSource] (System.Collections.Generic.IEnumerable`1[T] source, System.Threading.Tasks.ParallelOptions parallelOptions, System.Action`1[T] body) [0x0002a] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/Tasks/Parallel.cs:1748 
  at MonoDevelop.Ide.FindInFiles.FindReplace.FindAll (MonoDevelop.Ide.FindInFiles.Scope scope, MonoDevelop.Core.ProgressMonitor monitor, System.String pattern, System.String replacePattern, MonoDevelop.Ide.FindInFiles.FilterOptions filter, System.Threading.CancellationToken token) [0x00366] in /Users/vsts/agent/2.140.0/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.FindInFiles/FindReplace.cs:169 
  at MonoDevelop.Ide.FindInFiles.FindInFilesDialog+<>c__DisplayClass62_0.<SearchReplace>b__0 () [0x0009f] in /Users/vsts/agent/2.140.0/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.FindInFiles/FindInFilesDialog.cs:884 
```

> VS bug [#685793](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/685793)

**Comment by KirillOsenkov** [comment]
https://github.com/mono/monodevelop/blob/f90135d38af0b223174b872c448f0114e1bdb4c6/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.FindInFiles/FindReplace.cs#L169

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.8.0.537 (master)

Pull Request #6032 merged by: Lluis Sanchez
Author: mono
Commit: 477b9d1d53b00f040b7dc3d59709784ae9ea746d (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.7.0.1377 (release-7.7)

Pull Request #6033 merged by: Lluis Sanchez
Author: mono
Commit: 43af652717503569f3c73a9a4f9a6f4449898f1b (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.6.0.802 (master)

Pull Request #6032 merged by: Lluis Sanchez
Author: mono
Commit: 477b9d1d53b00f040b7dc3d59709784ae9ea746d (xamarin/monodevelop)
